Sources/Workflow/Models/Workflow.swift

```diff
@@ -191,7 +192,8 @@ public final class Workflow<F: FlowRepresentable>: LinkedList<_WorkflowItem> {
         }
 
         guard let nextNode = nextLoadedNode else {
-            onFinish?(argsToPass)
+            #warning("Workflow in Complete is incorrect. Test Drive it to make it correct.")
+            orchestrationResponder?.complete(AnyWorkflow(Workflow(self.last)), passedArgs: argsToPass, onFinish: onFinish)
```

Comment:
Ok so this is a little odd.  Because I used a node from the linked list the first time, we ended up with a convoluted `AnyWorkflow(self)` which is correct.  Then this code is now actually wrong, BUT it won't fail tests because it still falls under what we are testing (the last item is persisted or not).

So we're still kind of right but really we either need to add tests or finally set it completely right with:
```suggestion
            orchestrationResponder?.complete(AnyWorkflow(self), passedArgs: argsToPass, onFinish: onFinish)
```

Reply:
We added a test.